enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+### Workflow Description
+
+```mermaid
+sequenceDiagram
+actor Admin
+participant Installer
+participant Bootstrap
+participant CVO
+participant ClusterOperator
+participant PO manager
+Installer->>Admin: Creates the Kubernetes manifests
+Installer->>PO custom resource: Generates
+Admin-->>Installer: Edits Kubernetes manifests
+Installer->>Bootstrap: Creates
+Bootstrap->>CVO: Deploys
+CVO->>PO manager: Deploys
+PO manager->>ClusterOperator: Manages
+PO manager->>PO custom resource: Watches
+CVO->>ClusterOperator: Watches
+```
+
+Note: The above workflow requires admins to manually create, and edit, the cluster Kubernetes manifests with their desired PO intent. There's a chance that admins mistype the package names, and depending on whether upfront validation is possible, that invalid configuration may not be surfaced until after cluster creation. This behavior may require explicit documentation to reduce this risk. These are obvious UX gaps with this workflow, which can be solidified in future phases, but it will result in zero installer changes during the 4.12 timeframe.
+
+### API Extensions
+
+- Adds a top-level PlatformOperator cluster-scoped CRD.
+- Adds the Bundle and BundleDeployment OLM 1.x API(s).
+- Adds a validating webhook for the Bundle and BundleDeployment APIs.
+- Adds a CRD validation webhook.
+
+### Risks and Mitigations
+
+- Early phases will introduce multiple v1alpha1 CRDs downstream.
+- Early phases will introduce OLM v1 components that are in alpha stages.
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions that may require upfront RBAC to limit this API to admins.
+
+### Drawbacks
+
+- Implementing this PO mechanism using legacy OLM would provide more immediate business value. That implementation is outlined below in the alternatives section.
+- Early phases will be delivered through tech preview guidelines. This leads to little customer feedback, given it precludes production clusters.
+- Early phases may introduce a new OLM 1.x control plane that has little-to-no interaction with the existing, legacy OLM 0.x control plane.
+- Early phases may introduce validating webhooks. These webhooks can impact the cluster's overall health when they're not available.
+- Early phases may introduce validating webhooks. By default, these webhooks will be deployed using an HA configuration, but may not have logic to accommodate SNO cluster environments.
+- Early phases may introduce a dynamic watching implementation for managing the underlying OLM operator resources. This has potential performance implications that need to be measured through benchmarking and/or addressed before any GA phase.
+- Future phase designs may require a large-scale rewrite of the OLM component.
+
+## Design Details
+
+### Open Questions [optional]
```

Comment:
For components that are commonly used, but would become optional under this plan, what would the experience be for mirroring them into a disconnected environment? If I mirror a release image in the future, will that include less stuff than today, and I'll have to take extra steps to mirror the optional components I want?

Reply:
> For components that are commonly used, but would become optional under this plan, what would the experience be for mirroring them into a disconnected environment?

you'd have to do oc adm catalog mirroring, as you do for any other olm operator you want to install in a disconnected environment today.

>  If I mirror a release image in the future, will that include less stuff than today, and I'll have to take extra steps to mirror the optional components I want?

once we get to the point of actually moving things that are currently in the payload, out of it, yes.  (the first candidates to be POs are things that are just regular olm operators today.  Migrating something from CVO-operator to PO will come later)

Reply:
Added this thread into the answered questions section. Updated some of the summary/motivation sections noting that we may require migration in the future but that will happen in future iterations of composable openshift.